src/lib/filter.ts

```diff
@@ -22,6 +22,20 @@ export function shouldSkipRequest(binding: Binding, req: AdmissionRequest, capab
   const { metadata } = srcObject || {};
   const combinedNamespaces = [...namespaces, ...capabilityNamespaces];
 
+  // Delete bindings do not work through admission with DeletionTimestamp
+  if (binding.event.includes(Event.Delete) && binding.filters?.deletionTimestamp) {
+    return true;
+  }
+
+  // Test for deletionTimestamp
+  if (
+    binding.filters &&
+    binding.filters.deletionTimestamp &&
+    req.object.metadata &&
+    !req.object.metadata?.deletionTimestamp
```

Comment:
Seeing a code-style trend for checking `X.prop` and then drilling into `X.prop.subprop` that might also be common in Pepr or provides some benefit I'm unaware of compared to `?` for chaining